Jlr

MZorzy opened the pull request:
add few sensor dorr position/lock,
home fix for hass.io rederence under custom_components 
warning: tyre pressure is like "2700" from RRS tdi BUT "270" from RRS HST.

MZorzy committed:
Merge pull request #1 from rb28z2/jlr

Jlr

MZorzy committed:
Update __init__.py

add door lock/position sensor, tyre pressure, alarm status

MZorzy committed:
Update sensor.py

fix reference for hass.io, add theftalarm, int odometer

MZorzy committed:
Update binary_sensor.py

fix for hass.io

MZorzy committed:
Add files via upload

jlrpy.py 1.2.0

rb28z2 commented on homeassistant/components/jlrincontrol/__init__.py:
```diff
@@ -35,13 +35,26 @@
     'DISTANCE_TO_EMPTY_FUEL': ('sensor', 'Range', 'mdi:road', 'km'),
-    'EXT_KILOMETERS_TO_SERVICE': ('sensor', 'Distance to next service',
-                                  'mdi:garage', 'km'),
+    'EXT_KILOMETERS_TO_SERVICE': ('sensor', 'Distance to next service', 'mdi:garage', 'km'),
```
So this line was split into 2 because of home assistant upstream's linting requirements that lines must be less than 80 characters long. A little annoying yes, but its part of their code quality requirements so not a whole lot can be done about that. I'd check your entire PR to make sure everything is under that limit.

MZorzy replied:
oh, thank you.

rb28z2 commented on homeassistant/components/jlrincontrol/binary_sensor.py:
```diff
@@ -2,7 +2,7 @@
 
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.components.jlrincontrol import JLREntity, RESOURCES
+from . import JLREntity, RESOURCES
```
I'd check if this is allowed as per HomeAssistant requirements. Obviously syntactically its fine, but it might not be with regards to linting reqs.

MZorzy replied:
i see "relative import" with dot undet great migration here:
https://developers.home-assistant.io/blog/

MZorzy replied:
dot on reference like volvo
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/volvooncall/sensor.py
or bmw
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/bmw_connected_drive/device_tracker.py

rb28z2 replied:
Ah ok perfect. Looks like it's allowed them 🙂

rb28z2 commented on homeassistant/components/jlrincontrol/jlrpy.py:
```diff
@@ -0,0 +1,517 @@
+""" Simple Python class to access the JLR Remote Car API
```
This file shouldn't be in here as its not required. HomeAssistant downloads the required dependencies as required on bootup. Check the `requirements.txt` file IIRC.

rb28z2 commented on homeassistant/components/jlrincontrol/sensor.py:
```diff
@@ -1,11 +1,11 @@
 )
 
-from homeassistant.components.jlrincontrol import JLREntity, RESOURCES
+from . import JLREntity, RESOURCES
```
Same as above. This shortcut might not be allowed.

rb28z2 reviewed (changes requested):
A few things need to be addressed, but overall it looks good.

I strongly suggest running HomeAssistant's suite of tests locally and make sure they pass.
https://developers.home-assistant.io/docs/en/development_testing.html

MZorzy committed:
Delete jlrpy.py

MZorzy committed:
Update __init__.py

isort

MZorzy committed:
Add files via upload

Blak reformatted
https://developers.home-assistant.io/docs/en/development_guidelines.html

MZorzy committed:
Update sensor.py

MZorzy committed:
Update __init__.py
